core/Updates/4.0.0-b1.php

```diff
@@ -104,6 +106,19 @@ public function getMigrations(Updater $updater)
         $migrations[] = $this->migration->db->dropColumn('user', 'alias');
         $migrations[] = $this->migration->db->dropColumn('user', 'token_auth');
 
+        // remove outdated sessions to avoid useless updates
+        Db::exec('DELETE FROM ' . Common::prefixTable('session') . ' WHERE modified + lifetime < ' . time());
+
+        $salt = SettingsPiwik::getSalt();
+        $sessions = Db::fetchAll('SELECT id from ' . Common::prefixTable('session'));
+
+        foreach ($sessions as $session) {
+            $migrations[] = $this->migration->db->boundSql(
```

Comment:
this would actually print sessionId hashes on the screen... of course they are hashed but it may still not be ideal. Matomo may be in that state for a while and people could see them ... it's very low risk though that it 1) becomes visible and 2) someone either has the salt or tries to brute force them (eg a sys admin who is updating Matomo could have the salt but no access to the DB and now could technically brute force the original session id) . If someone was to downgrade later to Matomo 3 then it become an actual sessionId though. 

![image](https://user-images.githubusercontent.com/273120/100386729-1e304780-308b-11eb-853b-af48d9fe35b7.png)

I wonder should we maybe only do this for session Ids that are > 190 characters? In all other cases we could simply log out users? and these we could even simply delete?

Or we simply execute the query in the background similar to the delete from session.

Reply:
in any case we should not do this for any 128 character session as this indicates the sessionId might be already hashed